### ISSUE TITLE
Static-init order dependence in quantized custom-class registration: PR #90133's fix covers only some consumer TUs, so `import torch` can abort under LTO

### ISSUE THREAD

**Issue by BwL1289** [pytorch/pytorch#192669]
`torch.ops.quantized.*` registration depends on cross-TU dynamic-initialization order. PR #90133
(2022) fixed this for some translation units by calling `register_conv_params<N>()` /
`register_linear_params()` / `register_embedding_params()` as the first statement of the consumer's
`TORCH_LIBRARY_IMPL` block, the construct-on-first-use idiom. Seven consumer TUs were not
converted, and they still infer schemas over `c10::intrusive_ptr<...PackedParamsBase>` before the
registrar TU has populated c10's custom-class registry.

With a default (non-LTO) build the surviving order happens to be benign, so this is invisible. Give
the build a genuine cross-TU view, ThinLTO plus `-fwhole-program-vtables` in our case, and
`.init_array` is reordered, the consumer runs first, and a bare `import torch` aborts at load
time.

Why this is a real defect, not a miscompile

Cross-TU dynamic initialization order is indeterminately sequenced. [basic.start.dynamic]/3.3.
It is unspecified, not undefined. A conforming toolchain is free to choose any order, so any
code that depends on one is relying on unspecified behaviour. The reordering that exposed it is
legal; the ordering dependence is the bug.

This distinction matters for triage: nothing here is a compiler bug, and no compiler flag is
"at fault". The same latent dependence is one link-order or LTO-decision away from firing on any
build configuration.

Affected translation units

All under `aten/src/ATen/native/`:

1. `quantized/cpu/qconv_dynamic.cpp`: `TORCH_LIBRARY_IMPL(quantized, CPU, m)`
2. `quantized/cpu/qconv.cpp`: `TORCH_LIBRARY_IMPL(quantized, QuantizedCPU, m)`
3. `quantized/cpu/qconv_prepack.cpp`
4. `quantized/cpu/qembeddingbag.cpp`
5. `quantized/cpu/qembeddingbag_prepack.cpp`
6. `quantized/cpu/qembeddingbag_unpack.cpp`
7. `ao_sparse/quantized/cpu/qlinear_dynamic.cpp`

Several also need `#include <ATen/native/quantized/library.h>` for the `register_*_params`
declarations,`qconv.cpp` and `qconv_prepack.cpp` do not currently include it.

Proposed fix upstream's own idiom applied to the TUs PR #90133 skipped

```cpp
TORCH_LIBRARY_IMPL(quantized, CPU, m) {
  register_conv_params<2>();     // idempotent; construct-on-first-use
  register_conv_params<3>();
  // ... existing registrations
}
```

This is preferable to pinning an order (e.g. `[[gnu::constructor(101)]]`) because it removes the
order dependence rather than selecting one, and it converges these TUs with the ones PR #90133
already converted. (Note `[[gnu::init_priority(N)]]` does not compile in this position.)

Reproduction

Any build that reorders `.init_array` such that a consumer TU above precedes the registrar. We hit
it with clang ThinLTO + `-fwhole-program-vtables`; the abort is at `import torch`, before any user
code runs. It can also be demonstrated without changing flags by reversing `.init_array` order in
the produced `libtorch_cpu.so`.

What we verified

- With the seven TUs converted, `import torch` succeeds and
  `torch.ops.quantized.conv1d_dynamic` / `linear_dynamic` resolve.
- A full 663-entry `.init_array` reversal imports cleanly with the fix applied.
- Demoting the registrar TU to last, and separately demoting `library.cpp` (which removes an
  accidental ordering that was masking four of the consumers), both import cleanly with the fix.

cc @malfet @zklaus @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel
